test/powershell/Host/TabCompletion/TabCompletion.Tests.ps1

```diff
@@ -84,6 +84,17 @@ Describe "TabCompletion" -Tags CI {
         $res.CompletionMatches[0].CompletionText | Should -BeExactly '$CurrentItem'
     }
 
+    It 'Should complete variables set with an attribute' {
+        $res = TabExpansion2 -inputScript '[ValidateNotNull()]$Var1 = 1; $Var'
+        $res.CompletionMatches[0].CompletionText | Should -BeExactly '$Var1'
+    }
+
+    It 'Should use the first type constraint in a variable assignment in the tooltip' {
+        $res = TabExpansion2 -inputScript '[int] [string] $Var1 = 1; $Var'
+        $res.CompletionMatches[0].CompletionText | Should -BeExactly '$Var1'
+        $res.CompletionMatches[0].ToolTip | Should -BeExactly '[int]$Var1'
+    }
```

Comment:
@MartinGC94 Both tests works for me without the PR. Can you clarify that we fix here?

Reply:
You most likely have `$var1` defined in your session state. Try opening a fresh session with -NoProfile and try to rerun the tests.

Reply:
From fresh
![image](https://github.com/user-attachments/assets/a1b78ee5-a205-4849-be26-8916f729e5de)

Reply:
Doesn't happen in retail builds:
![image](https://github.com/user-attachments/assets/3e4b5f35-aca7-42d6-817d-4707414974cd)

Reply:
Ah, I see. It would be great to understand where the regression come from. (Then maybe add more test to exclude such regressions.)

Reply:
It must have come from this old PR: https://github.com/PowerShell/PowerShell/pull/19595 where I rewrote the ast visitor. The issue was caused by the ast visitor not handling attributed expressions in assignments, and the implementation for type constrained assignments was naive and assumed no other attributes was added. Now there's tests covering these new scenarios so a regression shouldn't happen again.

Reply:
Thanks for sharing the history. Will look soon.